src/main/java/com/iexec/core/task/update/TaskUpdatePriorityBlockingQueue.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2021 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.core.task.update;
+
+import com.iexec.core.task.Task;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.PriorityBlockingQueue;
+import java.util.stream.Stream;
+
+public class TaskUpdatePriorityBlockingQueue extends PriorityBlockingQueue<Runnable> {
```

Comment:
Was it not possible to use `PriorityBlockingQueue<TaskUpdate>` with `TaskUpdate` extending `Runnable` ?
This would allow to reuse all the methods from PriorityBlockingQueue like `contains`, `stream` which would return a `Stream<TaskUpdate>` in our scope.

Reply:
The only method without the `@Override` annotation is only used in tests.

Reply:
Sadly, that's not how templates work in Java. A `PriorityBlockingQueue<? extends Runnable>` is not compatible with a `PriorityBlockingQueue<Runnable>`.

Regarding the method you're talking about, there are 3 non-annotated methods: `takeAll`, `streamAsTaskUpdate` and `containsTask`. I think you're talking about `takeAll`, right? I thought about putting it into the test class but I was not sure about this design. Wouldn't that method be handy someday?

Reply:
My main point was on `PriorityBlockingQueue<TaskUpdate>`.
If it's not possible to use this, my second comment becomes irrelevant and we can keep the new methods. :+1: